src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py

```diff
@@ -243,12 +245,19 @@ def export_config(cmd,
     src_kvs = __read_kv_from_config_store(azconfig_client,
                                           key=key,
                                           label=label if label else SearchFilterOptions.EMPTY_LABEL,
-                                          prefix_to_remove=prefix,
+                                          prefix_to_remove=prefix if not export_as_reference else "",
                                           cli_ctx=cmd.cli_ctx if resolve_keyvault else None)
 
     if skip_keyvault:
         src_kvs = [keyvalue for keyvalue in src_kvs if keyvalue.content_type != KeyVaultConstants.KEYVAULT_CONTENT_TYPE]
 
+    if destination == 'appservice' and export_as_reference:
```

Comment:
We can move this block after we have invoked `__discard_features_from_retrieved_kv`. That's because we dont want to call `__map_to_appservice_config_reference` for any feature flags.